tls/s2n_resume.c

```diff
@@ -868,19 +869,22 @@ int s2n_decrypt_session_ticket(struct s2n_connection *conn, struct s2n_stuffer *
     POSIX_GUARD(s2n_stuffer_skip_write(&state_stuffer, state_blob_size));
     POSIX_GUARD_RESULT(s2n_deserialize_resumption_state(conn, &from->blob, &state_stuffer));
 
-    uint64_t now = 0;
-    POSIX_GUARD_RESULT(s2n_config_wall_clock(conn->config, &now));
+    if (s2n_connection_get_protocol_version(conn) == S2N_TLS13) {
+        return S2N_SUCCESS;
+    }
 
-    /* If the key is in decrypt-only state, then a new key is assigned
-     * for the ticket.
+    /* A new key is assigned for the ticket if the key is in decrypt-only state, 
+     * a key in encrypt-decrypt state is available, and TLS 1.2 is being used.
      */
-    if (now >= key->intro_timestamp + conn->config->encrypt_decrypt_key_lifetime_in_nanos) {
-        /* Check if a key in encrypt-decrypt state is available */
-        if (s2n_config_is_encrypt_decrypt_key_available(conn->config) == 1) {
-            conn->session_ticket_status = S2N_NEW_TICKET;
-            POSIX_GUARD_RESULT(s2n_handshake_type_set_tls12_flag(conn, WITH_SESSION_TICKET));
-            return S2N_SUCCESS;
-        }
+    uint64_t now = 0;
```

Comment:
Okay I actually changed my mind, I actually think the nested if-statement is more readable. Can you change it to:
```
if (ticket key is in decrypt mode) {
     if (an encryption key is available) {
         set tls12 flag
    }
}
```
Also this is a nit but you don't actually need to call `return S2N_SUCCESS` in the if-statement. Not sure why that was originally coded that way.